### ISSUE TITLE
fetch is not a function

### ISSUE THREAD

**Issue by skstronghold** [firebase/firebase-js-sdk#75]
### Describe your environment

  * Operating System version:  Ubuntu 14.04.02
  * Firebase SDK version: 4.1.3
  * Firebase Product: messaging

### Describe the problem
Firebase messaging doesn't work with latest npm package. It throws:
```
TypeError: fetch is not a function
    at WindowController.manifestCheck_ (window-controller.js?:149)
    at WindowController.getToken (window-controller.js?:125)
```
Exact line in code is [here](https://github.com/firebase/firebase-js-sdk/blob/c054dab68ce4d30af7fcb0bb850f1a064ba023c5/src/messaging/controllers/window-controller.ts#L119)

#### Steps to reproduce:

Install latest firebase package from npm. Follow tutorial given [here](https://firebase.google.com/docs/cloud-messaging/js/client).
#### Relevant Code:
```javascript
      firebase.initializeApp(opts);
      const messaging = firebase.messaging();
      messaging.requestPermission()
        .then(function() {
          console.log('Notification permission granted.');
          return messaging.getToken()
            .then((currentToken) => {
              if (currentToken) {
                console.log("Token is: " + currentToken)
              } else {
                // you don't have permission to show notifications
                // detect whether they are blocked or not, then show your custom UI
              }
            })
            .catch((err) => {
              console.log("Error to get token is: " + err);
              // retrieving token failed, analyze the error
            });
        })
        .catch(function(err) {
          console.log('Unable to get permission to notify.', err);
        });
```

**Comment by jsayol** [comment]
It seems like there should be a check for the Fetch API [here](https://github.com/firebase/firebase-js-sdk/blob/c054dab68ce4d30af7fcb0bb850f1a064ba023c5/src/messaging/controllers/window-controller.ts#L360-L367).

**Comment by jshcrowthe** [comment]
@skstronghold On which browser are you seeing this behavior? Messaging only supports browsers that support service worker (see https://firebase.google.com/docs/cloud-messaging/js/client) so I'd be curious if this isn't a browser issue.

That said, @jsayol is probably right that we should add that check.

**Comment by gauntface** [comment]
The SDK should perform an additional check here: https://github.com/firebase/firebase-js-sdk/blob/c054dab68ce4d30af7fcb0bb850f1a064ba023c5/src/messaging/controllers/window-controller.ts#L360

**Comment by jshcrowthe** [comment]
We have merged #79 at it will go out in the next release. Thanks for the issue @skstronghold
